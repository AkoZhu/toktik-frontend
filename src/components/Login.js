import * as React from 'react';
import Button from '@mui/material/Button';
import CssBaseline from '@mui/material/CssBaseline';
import TextField from '@mui/material/TextField';
import FormControlLabel from '@mui/material/FormControlLabel';
import Checkbox from '@mui/material/Checkbox';
import Link from '@mui/material/Link';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';
import {createTheme, ThemeProvider} from '@mui/material/styles';
import logo from "../assets/logo.png";
import {useState} from "react";
import Home from "./Home";
import {Paper} from "@mui/material";

function Copyright(props) {
    return (
        <Typography variant="body2" color="text.secondary" align="center" {...props}>
            {'Copyright Â© '}
            <Link color="inherit" href="https://toktik.com/">
                TokTik
            </Link>{' '}
            {new Date().getFullYear()}
            {'.'}
        </Typography>
    );
}

function LoginComponent() {
    const [ToHome, setToHome] = useState(false)


    const handleSubmit = (event) => {
        event.preventDefault();
        const data = new FormData(event.currentTarget);
        setToHome(true)
        console.log({
            email: data.get('email'),
            password: data.get('password'),
        });
    };

    if(ToHome){
        console.log(ToHome)
        return (
            <Home/>
        )
    }else

    return (
        <ThemeProvider theme={theme}>
            <Grid container component="main" sx={{ height: '90vh', mt: "30px" }}>
                <CssBaseline />
                <Grid
                    item
                    xs={false}
                    sm={4}
                    md={7}
                    sx={{
                        backgroundImage: 'url(https://source.unsplash.com/random)',
                        backgroundRepeat: 'no-repeat',
                        backgroundColor: (t) =>
                            t.palette.mode === 'light' ? t.palette.grey[50] : t.palette.grey[900],
                        backgroundSize: 'cover',
                        backgroundPosition: 'center',
                    }}
                />
                <Grid item xs={12} sm={8} md={5} component={Paper} elevation={6} square>
                    <Box
                        sx={{
                            my: 8,
                            mx: 4,
                            display: 'flex',
                            flexDirection: 'column',
                            alignItems: 'center',
                        }}
                    >
                        <img src={logo} className="logo" alt="logo" style={{width: '288x', height: '120px'}}/>
                        <Typography component="h1" variant="h5">
                            Sign in
                        </Typography>
                        <Box component="form" noValidate onSubmit={handleSubmit} sx={{ mt: 1 }}>
                            <TextField
                                margin="normal"
                                required
                                fullWidth
                                id="email"
                                label="Email Address"
                                name="email"
                                autoComplete="email"
                                autoFocus
                            />
                            <TextField
                                margin="normal"
                                required
                                fullWidth
                                name="password"
                                label="Password"
                                type="password"
                                id="password"
                                autoComplete="current-password"
                            />
                            <FormControlLabel
                                control={<Checkbox value="remember" color="primary" />}
                                label="Remember me"
                            />
                            <Button
                                type="submit"
                                fullWidth
                                variant="contained"
                                sx={{ mt: 3, mb: 2 }}
                            >
                                Sign In
                            </Button>
                            <Grid container>
                                <Grid item xs>
                                    <Link href="#" variant="body2">
                                        Forgot password?
                                    </Link>
                                </Grid>
                                <Grid item>
                                    <Link href={"/signup"} variant="body2">
                                        {"Don't have an account? Sign Up"}
                                    </Link>
                                </Grid>
                            </Grid>
                        </Box>
                    </Box>
                </Grid>
            </Grid>
        </ThemeProvider>
    )
}

const theme = createTheme();

export default function Login() {

    return (
        <ThemeProvider theme={theme}>
            <Container component="main" fixed>
                <CssBaseline/>
                <LoginComponent/>
                <Copyright sx={{mt: 8, mb: 4}}/>
            </Container>
        </ThemeProvider>
    );
}