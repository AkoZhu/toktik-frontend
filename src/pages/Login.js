import * as React from 'react';
import {useState} from 'react';
import Button from '@mui/material/Button';
import CssBaseline from '@mui/material/CssBaseline';
import TextField from '@mui/material/TextField';
import FormControlLabel from '@mui/material/FormControlLabel';
import Checkbox from '@mui/material/Checkbox';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';
import logo from "../assets/logo.png";
import {DialogActions, DialogContent, DialogContentText, DialogTitle, Paper} from "@mui/material";
import SignUpModal from "../components/login/SignUpModal";
import theme from "../theme";
import {ThemeProvider} from "@mui/material/styles";
import {login} from "../api/login";
import Dialog from '@mui/material/Dialog';
import LogoutIcon from "@mui/icons-material/Logout";

import {Navigate} from "react-router-dom";


export default function Login() {
    React.useEffect(() => {
        sessionStorage.clear();
    }, [])

    return (
        <ThemeProvider theme={theme}>
            <Container component="main" fixed>
                <CssBaseline/>
                <LoginComponent/>
            </Container>
        </ThemeProvider>
    );
}

export function LogoutDialog(){
    const [open, setOpen] = React.useState(false);

    const handleClickOpen = () => {
        setOpen(true);
    };


    const handleClose = () => {
        setOpen(false);
    };

    const handleLogout = () => {
        handleClose();
        sessionStorage.clear();
        window.location.reload();
    }

    return (
        <div>
            <LogoutIcon variant="outlined" onClick={handleClickOpen} sx ={{position:"relative", top:-2}}/>
            <Dialog
                open={open}
                onClose={handleClose}
                aria-labelledby="alert-dialog-title"
                aria-describedby="alert-dialog-description"
            >
                <DialogTitle id="alert-dialog-title">
                    {"Logout confirmation"}
                </DialogTitle>
                <DialogContent>
                    <DialogContentText id="alert-dialog-description">
                        Are you sure you want to logout?
                    </DialogContentText>
                </DialogContent>
                <DialogActions>
                    <Button onClick={handleClose}>Cancel</Button>
                    <Button onClick={handleLogout} color="error"  autoFocus>
                        Confirm
                    </Button>
                </DialogActions>
            </Dialog>
        </div>
    );
}



function LoginComponent() {
    const [ToFeed, setToFeed] = useState(false);

    const handleSubmit = (event) => {
        event.preventDefault();
        const data = new FormData(event.currentTarget);

        login(data.get('username'), data.get('password')).then((success) => {
            if (success) {
                setToFeed(true);
            } else {
                alert("Wrong username or password")
            }
        });
    };

    if (ToFeed) {
        return (<Navigate to={'/'}/>);
    }


    return (
        <ThemeProvider theme={theme}>
            <Grid container component="main" sx={{height: '90vh', mt: "30px"}}>
                <CssBaseline/>
                <Grid
                    item
                    xs={false}
                    sm={4}
                    md={7}
                    sx={{
                        backgroundImage: 'url(https://source.unsplash.com/random)',
                        backgroundRepeat: 'no-repeat',
                        backgroundColor: (t) =>
                            t.palette.mode === 'light' ? t.palette.grey[50] : t.palette.grey[900],
                        backgroundSize: 'cover',
                        backgroundPosition: 'center',
                    }}
                />
                <Grid item xs={12} sm={8} md={5} component={Paper} elevation={6} square>
                    <Box
                        sx={{
                            my: 8,
                            mx: 4,
                            display: 'flex',
                            flexDirection: 'column',
                            alignItems: 'center',
                        }}
                    >
                        <img src={logo} className="logo" alt="logo" style={{width: '288x', height: '120px'}}/>
                        <Typography component="h1" variant="h5">
                            Sign in
                        </Typography>
                        <Box id="loginForm" component="form" noValidate onSubmit={handleSubmit} sx={{mt: 1}}>
                            <TextField
                                margin="normal"
                                required
                                fullWidth
                                id="username"
                                label="Username"
                                name="username"
                                autoComplete="username"
                                autoFocus
                            />
                            <TextField
                                margin="normal"
                                required
                                fullWidth
                                name="password"
                                label="Password"
                                type="password"
                                id="password"
                                autoComplete="current-password"
                            />
                            <FormControlLabel
                                control={<Checkbox value="remember" color="primary"/>}
                                label="Remember me"
                            />
                            <Button
                                type="submit"
                                fullWidth
                                variant="contained"
                                sx={{mt: 3, mb: 2}}
                                aria-label="button-signIn"
                            >
                                Sign In
                            </Button>
                        </Box>
                        <Grid container>
                            <Grid item xs={6}>
                                <Button>
                                    Forgot password?
                                </Button>
                            </Grid>
                            <Grid item xs={6}>
                                <SignUpModal ToFeed={ToFeed} handleToFeed={setToFeed}/>
                            </Grid>
                        </Grid>
                    </Box>
                </Grid>
            </Grid>
        </ThemeProvider>
    )
}
