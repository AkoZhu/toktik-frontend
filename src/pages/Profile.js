import React from "react";
import Layout from "../components/common/Layout";
import ProfilePicture from "../components/profile/ProfilePicture";
import {Card, CardContent} from "@mui/material";
import {useParams} from "react-router-dom";
import ProfileTabs from "../components/profile/ProfileTabs";
import Box from "@mui/material/Box";
import theme from "../theme"
import LoadingScreen from "../components/common/LoadingScreen";
import {styles, useProfilePageStyles} from "../components/profile/ProfileStyle";
import {ProfileNameSection} from "../components/profile/ProfileNameSection";
import {PostCountSection} from "../components/profile/PostCountSection";
import {getFollowerCountByUsername, getUserByName} from "../api/user";
import {getPostByUsername} from "../api/post";

function Profile() {
    let params = useParams();
    const currentUser = sessionStorage.getItem("CurrentUsername");
    const profileUsername = params.username ? params.username : currentUser;

    const isOwner = profileUsername === currentUser;
    const useStyles = useProfilePageStyles(theme);

    const [user, setUser] = React.useState({});
    const [posts, setPosts] = React.useState([]);
    const [followerCount, setFollowerCount] = React.useState();


    React.useEffect(() => {
        async function fetchData() {
            setUser(await getUserByName(profileUsername));
            setPosts(await getPostByUsername(profileUsername));
            setFollowerCount(await getFollowerCountByUsername(profileUsername));
        }

        fetchData().then(() => {
        });
    }, [profileUsername]);

    function handleFollower() {
        getFollowerCountByUsername(profileUsername).then((data) => {
            setFollowerCount(data);
        });
    }

    if (!currentUser || !user) return <LoadingScreen/>;

    return (
        <Layout data-testid='profile_1'
                title={`${user.firstName} (@${user.username})`}
        >
            <Box component="div" sx={useStyles.container} data-testid='profile_2'>
                <Card sx={styles.cardLarge}>
                    <ProfilePicture image={user.profilePicture} isOwner={isOwner}/>
                    <CardContent sx={styles.cardContentLarge}>
                        <ProfileNameSection
                            user={user}
                            isOwner={isOwner}
                            handleFollower={handleFollower}
                            data-testid='profile_3'
                        />
                        <PostCountSection
                            username={profileUsername}
                            postCount={posts.length}
                            followerCount={followerCount}
                        />
                    </CardContent>
                </Card>
                <ProfileTabs user={user} posts={posts} isOwner={isOwner} data-testid='profile_4'/>
            </Box>
        </Layout>
    );
}

export default Profile;