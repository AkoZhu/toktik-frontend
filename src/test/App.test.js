/* eslint testing-library/prefer-screen-queries: "off", testing-library/no-await-sync-query: "off" */

import {fireEvent, render} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer, {act} from 'react-test-renderer';
import Login from "../pages/Login.js";
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import FeedPage from "../pages/Feed";
import {BrowserRouter} from 'react-router-dom';
import Profile from "../pages/Profile";

const mockAxios = new MockAdapter(axios.create());

async function mockLogin() {
    const response = await axios.get('http://localhost:4000/login');
    return response.data;
}

function mockSessionStorage() {
    sessionStorage.setItem("CurrentUsername", "demo");
    sessionStorage.setItem("CurrentUserId", "1");
}

describe("login", () => {

    test('renders login page', () => {
        const view = render(<Login/>);
        const inputNode = view.getByText("Username");
        const inputNode2 = view.getByText("Password");
        expect(inputNode).toBeInTheDocument();
        expect(inputNode2).toBeInTheDocument();
    });

    test('snapshot', () => {
        const component = renderer.create(<Login/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('username input should accept text', () => {
        const view = render(<Login/>);
        const inputNode = view.getByLabelText(/^Username/i);
        expect(inputNode.value).toMatch("");
        fireEvent.change(inputNode, {target: {value: "Username"}});
        expect(inputNode.value).toMatch("Username");


    });

    test('password input should accept text', () => {
        const view = render(<Login/>);
        const inputNode = view.getByLabelText(/^password/i);
        expect(inputNode.value).toMatch("");
        fireEvent.change(inputNode, {target: {value: "password"}});
        expect(inputNode.value).toMatch("password");


    });


    test("sessionStorage", () => {
        const username = "demo";
        mockSessionStorage();
        expect(sessionStorage.getItem("CurrentUsername")).toBe(username);
    })


    test("redirect to feed page", async () => {
        const view = render(<Login/>, {wrapper: BrowserRouter})
        const user = userEvent.setup();

        // verify page content for default route
        expect(view.getByLabelText(/^Username/i)).toBeInTheDocument()

        const usernameNode = view.getByLabelText(/^Username/i);
        fireEvent.change(usernameNode, {target: {value: "demo"}});
        const pwdNode = view.getByLabelText(/^password/i);
        fireEvent.change(pwdNode, {target: {value: "123456"}});

        // verify page content for expected route after navigating
        await user.click(view.getByRole("button", {name: /button-signIn/i}))

    })
});

describe('the signup api should return success', () => {
    // seed data for all get requests. You can specify an endpoint to mock
    test('signup async', async () => {
        const data = await mockLogin();
        expect(data.success).toBe(true);
    });
});

// Login Axios.
describe('login axios.', () => {
    // seed data for all get requests. You can specify an endpoint to mock
    mockAxios.onGet().reply(200, {
        success: true
    });

    test('check login api', async () => {
        const data = await axios.get("http://localhost:4000/login");
        expect(data.data.success).toBe(true);
    });
});

// Register Axios.
describe("register.", () => {
    // seed data for all get requests. You can specify an endpoint to mock
    mockAxios.onGet().reply(200, {
        success: true
    });

    test('check signup api', async () => {
        const data = await axios.get("http://localhost:4000/signup");
        expect(data.data.success).toBe(true);
    });

    test("render login", async () => {
        const view = render(<Login/>);

        const button = view.getByText("Don't have an account? Sign Up")
        fireEvent.click(button);

        // test Input
        const usernameNode = view.getByLabelText(/^Username/i);
        expect(usernameNode.value).toMatch("");
        fireEvent.change(usernameNode, {target: {value: "Username"}});
        expect(usernameNode.value).toMatch("Username");

        const firstNameNode = view.getByLabelText(/^First Name/i);
        expect(firstNameNode.value).toMatch("");
        fireEvent.change(firstNameNode, {target: {value: "FirstName"}});
        expect(firstNameNode.value).toMatch("FirstName");

        const lastNameNode = view.getByLabelText(/^Last Name/i);
        expect(lastNameNode.value).toMatch("");
        fireEvent.change(lastNameNode, {target: {value: "LastName"}});
        expect(lastNameNode.value).toMatch("LastName");

        const emailNode = view.getByLabelText(/^Email Address/i);
        expect(emailNode.value).toMatch("");
        fireEvent.change(emailNode, {target: {value: "emailName"}});
        expect(emailNode.value).toMatch("emailName");

        const passwordNode = view.getByLabelText(/^Password/i);
        expect(passwordNode.value).toMatch("");
        fireEvent.change(passwordNode, {target: {value: "Password"}});
        expect(passwordNode.value).toMatch("Password");

        const data = await axios.get("http://localhost:4000/signup");
        expect(data.data.success).toBe(true);

    })
})

describe("feed page", () => {
    test("render feed page", async () => {
        const view = render(<FeedPage/>);
        const inputNode = await view.getByRole("progressbar");
        expect(inputNode).toBeInTheDocument();
    })
    test('snapshot', async () => {
        const component = renderer.create(<FeedPage/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        await act(async () => render(<FeedPage/>));
    });
    test("render feed page", async () => {
        mockSessionStorage();
        const view = render(<FeedPage/>);
        setTimeout(async () => {
            const feed_1 = view.getByText("Load More");
            fireEvent.click(feed_1);
            await act(async () => render(<FeedPage/>));
        },1000)
    })
});

describe("profile page", () => {
    test("render profile page", async () => {
        const view = render(<Profile/>);
        const inputNode = await view.getByRole("progressbar");
        expect(inputNode).toBeInTheDocument();
    })
    test('snapshot', () => {
        const component = renderer.create(<Profile/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("check profile page element", async ()=> {
        mockSessionStorage();
        const view = render(<FeedPage/>);
        setTimeout(() => {
            const inputNode = view.getByRole("progressbar");
            const inputNode2 = screen.queryByTestId('profile_1');
            const profile_2 = screen.getByTestId('profile_2');
            const profile_3 = screen.getByTestId('profile_3');
            const profile_4 = screen.getByTestId('profile_4');
            expect(inputNode2).toBeInTheDocument();
            expect(inputNode).toBeInTheDocument();
            expect(profile_2).toBeInTheDocument();
            expect(profile_3).toBeInTheDocument();
            expect(profile_4).toBeInTheDocument();

        }, 1000)
    })
});







