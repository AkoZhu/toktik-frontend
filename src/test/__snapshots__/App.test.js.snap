// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login email input should accept text 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-1ss2ei0-MuiContainer-root"
>
  <main
    className="MuiGrid-root MuiGrid-container css-1kvxc64-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-7 css-bxq4xw-MuiGrid-root"
    />
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 css-gadse4-MuiPaper-root-MuiGrid-root"
    >
      <div
        className="MuiBox-root css-1y8ugea"
      >
        <img
          alt="logo"
          className="logo"
          src="logo.png"
          style={
            Object {
              "height": "120px",
              "width": "288x",
            }
          }
        />
        <h1
          className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Sign in
        </h1>
        <form
          className="MuiBox-root css-164r41r"
          id="loginForm"
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="username"
              id="username-label"
            >
              Username
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="username"
                autoFocus={true}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="username"
                name="username"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Username
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="password"
              id="password-label"
            >
              Password
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="current-password"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="password"
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="password"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Password
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <label
            className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
          >
            <span
              className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <input
                className="PrivateSwitchBase-input css-1m9pwf3"
                data-indeterminate={false}
                onChange={[Function]}
                type="checkbox"
                value="remember"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckBoxOutlineBlankIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
            >
              Remember me
            </span>
          </label>
          <button
            aria-label="button-signIn"
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            Sign In
          </button>
        </form>
        <div
          className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              Forgot password?
            </button>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          >
            <div>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                Don't have an account? Sign Up
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-ijul54-MuiTypography-root"
  >
    Copyright © 
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1cghntr-MuiTypography-root-MuiLink-root"
      href="https://toktik.com/"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      TokTik
    </a>
     
    2022
    .
  </p>
</main>
`;
